modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache

```diff
@@ -171,22 +203,20 @@ export class {{classname}} {
         if ({{paramName}}) {
         {{#isCollectionFormatMulti}}
             {{paramName}}.forEach((element) => {
-                {{#useHttpClient}}queryParameters = {{/useHttpClient}}queryParameters.append('{{baseName}}', <any>element);
+                {{#useHttpClient}}queryParameters = {{/useHttpClient}}this.addToHttpParams(queryParameters,
+                  <any>element, '{{baseName}}');
             })
         {{/isCollectionFormatMulti}}
         {{^isCollectionFormatMulti}}
-            {{#useHttpClient}}queryParameters = {{/useHttpClient}}queryParameters.set('{{baseName}}', {{paramName}}.join(COLLECTION_FORMATS['{{collectionFormat}}']));
+            {{#useHttpClient}}queryParameters = {{/useHttpClient}}this.addToHttpParams(queryParameters,
```

Comment:
I just noticed that this is a breaking change...
@wing328 @akehir can we still include it?

Reply:
or would there be a way to avoid a breaking change?

Reply:
Hhm, just to understand this: The generator is supporting OpenApi v2 (Swagger) _and_ v3? (Because according to https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md collectionFormat is only in v2.)

The meaning of the (old) line is that the given array is joined with some separator and then treated as string? If this is the case, it is possible leaf this as before. The patch is only about allowing (js) objects as query parameters.

Reply:
I think it should be fine now